src/Control/Email/Email.php

```diff
@@ -2,236 +2,181 @@
+    use Configurable;
+    use Extensible;
+    use Injectable;
 
-    /**
-     * @var array
-     * @config
-     */
-    private static $cc_all_emails_to = [];
+    private static string|array $send_all_emails_to = [];
 
-    /**
-     * @var array
-     * @config
-     */
-    private static $bcc_all_emails_to = [];
+    private static string|array $cc_all_emails_to = [];
 
-    /**
-     * @var array
-     * @config
-     */
-    private static $send_all_emails_from = [];
+    private static string|array $bcc_all_emails_to = [];
+
+    private static string|array $send_all_emails_from = [];
 
     /**
+     * The default administrator email address or array of [email => name]
      * This will be set in the config on a site-by-site basis
      * @see https://docs.silverstripe.org/en/4/developer_guides/email/#administrator-emails
-     *
-     * @config
-     * @var string|array The default administrator email address or array of [email => name]
-     */
-    private static $admin_email = null;
-
-    /**
-     * @var Swift_Message
-     */
-    private $swiftMessage;
-
-    /**
-     * @var string The name of the HTML template to render the email with (without *.ss extension)
      */
-    private $HTMLTemplate = null;
+    private static string|array $admin_email = '';
 
     /**
-     * @var string The name of the plain text template to render the plain part of the email with
+     * The name of the HTML template to render the email with (without *.ss extension)
      */
-    private $plainTemplate = null;
+    private string $HTMLTemplate = '';
 
     /**
-     * @var Swift_MimePart
+     * The name of the plain text template to render the plain part of the email with
      */
-    private $plainPart;
+    private string $plainTemplate = '';
 
     /**
-     * @var array|ViewableData Additional data available in a template.
+     * Additional data available in a template.
      * Used in the same way than {@link ViewableData->customize()}.
      */
-    private $data = [];
+    private array|ViewableData $data = [];
 
-    /**
-     * @var array
-     */
-    private $failedRecipients = [];
+    private array|ViewableData $dataAtLastRender = [];
+
+    private array $failedRecipients = [];
 
     /**
      * Checks for RFC822-valid email format.
      *
-     * @param string $address
-     * @return boolean
-     *
      * @copyright Cal Henderson <cal@iamcal.com>
      *    This code is licensed under a Creative Commons Attribution-ShareAlike 2.5 License
      *    http://creativecommons.org/licenses/by-sa/2.5/
      */
-    public static function is_valid_address($address)
+    public static function is_valid_address(string $address): bool
     {
-        $validator = new EmailValidator();
-        return $validator->isValid($address, new RFCValidation());
+        $validator = Injector::inst()->get(EmailValidator::class);
+        $rfcValidation = Injector::inst()->get(RFCValidation::class);
```

Comment:
Do these really need to be injected? I feel like having an extension point if people want to customise email validation would be more appropriate.
Besides which this change is out of scope of the PR's intended purpose.

Reply:
It's changing little but it's setting a precedent that we use injector any time we need a new instance of some arbitrary class - even if it doesn't use the `Injectable` trait and isn't something we're already using the injector for.
There has been pushback in the past using the injector everywhere even for things that _are_ mean to be explicitly injectble (see https://github.com/silverstripe/silverstripe-framework/issues/10269), so I'm hesitant to set a precedent like this, especially since I think changing this validation logic is _extremely_ edge case.

Reply:
At the end of the day, this change is unrelated to the PR and is likely to cause more discussion and hold up the PR more than it's worth. If you think this injection is worthwhile it's probably better off as a separate PR in this case.

Reply:
Reverted